cmd/proxy/Dockerfile

```diff
@@ -1,13 +1,30 @@
-FROM golang:1.11-rc
+FROM gobuffalo/buffalo:v0.12.6 as buffalo-builder
+
+RUN mkdir -p $GOPATH/src/github.com/gomods/athens/cmd/proxy
+WORKDIR $GOPATH/src/github.com/gomods/athens/cmd/proxy
+
+# this will cache the npm install step, unless package.json changes
+ADD cmd/proxy/package.json .
+ADD cmd/proxy/yarn.lock .
+RUN yarn install --no-progress
 
 RUN mkdir -p $GOPATH/src/github.com/gomods/athens
 WORKDIR $GOPATH/src/github.com/gomods/athens
 
-COPY . .
+ADD . .
+RUN cd cmd/proxy && buffalo build -s -o /bin/app
+
+
+FROM golang:1.11-rc-alpine
+RUN apk add --no-cache bash git openssh ca-certificates
```

Comment:
If you just use golang:1.11-rc image, then you wont need to install git. I think golang:1.11-rc is ubuntu whereas alpine is closer to scratch.

https://hub.docker.com/_/golang/

Reply:
Smaller is nice, but if we prefer Ubuntu I can change. Before it was straight Alpine. Thanks!

Reply:
yeah! I like smaller is nice and more secure :) 
I do not know what we prefer tbh! I will let @gomods/maintainers decide about that